tools/wpt/tests/test_testfiles.py

```diff
@@ -11,3 +12,32 @@ def test_getrevish_kwarg():
 def test_getrevish_implicit():
     with patch("tools.wpt.testfiles.branch_point", return_value=u"base"):
         assert testfiles.get_revish() == b"base..HEAD"
+
+
+def test_affected_testfiles():
+    manifest_json = {
+        "items": {
+            "crashtest": {
+                "a/b/c/foo-crash.html": [
+                    [ "a/b/c/foo-crash.html", {} ],
+                ]
+            }
+        },
+        "paths": {
+            "a/b/c/foo-crash.html": [
+                "acdefgh123456",
+                "crashtest",
+            ]
+        },
+        "url_base": "/",
+        "version": 7,
+    }
+    mfst = manifest.Manifest.from_json("/", manifest_json)
+    with patch("tools.wpt.testfiles.load_manifest", return_value=mfst):
+        # Dependent affected tests are determined by walking the filesystem,
```

Comment:
I briefly tried both factoring out the `os.walk` from `testfiles.affected_testfiles` and also using `mkdtmp`/etc to build an entire file structure, but both seemed pretty painful. If we really want to do one of those (or have some other idea) I'm open to it.